### PR TITLE
[MeshMap] Fetch RemoteUserPref Component

### PR DESCRIPTION
**Pull request by vineethvanga18**
**Description**

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/lib/store.js [hunk 1]
```diff
@@ -11,8 +11,8 @@ const initialState = fromJS({
   user: {},
```
Comment (leecalcote) [inline ui/lib/store.js:11]: let's keep the changes in the meshmap branch specific to meshmap. check-in any other changes to the master branch.

File: ui/components/RemoteUserPref.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
```
Comment (leecalcote) [inline ui/components/RemoteUserPref.js:1]: let's check this into the layer5labs/meshmap repo.

File: ui/components/visualize/MesheryVisualizeComponent.js [hunk 3]
```diff
@@ -14,9 +14,7 @@ import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup';
 import elementsJson from './Elements';
 import dagre from 'cytoscape-dagre';
 import clsx from 'clsx';
-import {
-  TopologyIcon,
-} from '@patternfly/react-icons';
+import { TopologyIcon } from '@patternfly/react-icons';
```
Comment (leecalcote) [inline ui/components/visualize/MesheryVisualizeComponent.js:17]: LGTM

File: ui/pages/userpreference.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import UserPreference from "../components/UserPreference";
+import RemoteUserPref from "../components/RemoteUserPref";
```
Comment (leecalcote) [inline ui/pages/userpreference.js:1]: This needs to go into `master`. 

// @utkarsh-pro

File: ui/package.json [hunk 5]
```diff
@@ -42,6 +42,7 @@
     "@material-ui/icons": "^4.11.2",
     "@material-ui/lab": "^4.0.0-alpha.57",
     "@material-ui/pickers": "^3.2.8",
+    "@paciolan/remote-component": "^2.10.2",
```
Comment (leecalcote) [inline ui/package.json:45]: Needs to go into `master`?

File: ui/remote-component.config.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
```
Comment (leecalcote) [inline ui/remote-component.config.js:1]: drop into `master`

File: ui/next.config.js [hunk 7]
```diff
@@ -16,4 +16,13 @@ module.exports = compose([
         '/404': { page: '/_error' },
       }
     }
-  }]);
\ No newline at end of file
+  },
```
Comment (leecalcote) [inline ui/next.config.js:19]: place into `master`

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by vineethvanga18** [comment]
@leecalcote Why should these be in master branch? These are meshmap specific right

**Comment by leecalcote** [comment]
@vineethvanga18 some of these belong in `master` because this PR is creating an extension point in Meshery itself.

**Comment by leecalcote** [comment]
Here is a link to the example provider package that will provide some insight on the subject - https://drive.google.com/file/d/1mQieb01yK1dhM11bC0zUF66YrXzMEKel/view?usp=sharing and http://docs.meshery.io/extensibility/providers#building-a-provider. @utkarsh-pro will be demo'ing work on this topic on tomorrow's call. 👍

**Comment by vineethvanga18** [comment]
@leecalcote ^^

File: ui/package.json (hunk 5 above)
Comment (leecalcote) [reply ui/package.json:45]: @vineethvanga18 how is the MeshMap user preference going to be remotely loaded if this utility component isn't in `master`?

File: ui/pages/userpreference.js (hunk 4 above)
Comment (leecalcote) [reply ui/pages/userpreference.js:1]: @vineethvanga18 if you don't create an extension point in `master`, how will you inject this plugin's (MeshMap's) user preference component?

File: ui/package.json (hunk 5 above)
Comment (vineethvanga18) [reply ui/package.json:45]: Yes these should be in master.
And I guess @utkarsh-pro already pushed these to master

Comment (leecalcote) [reply ui/package.json:45]: @vineethvanga18 ready to check-in MeshMap user pref into Meshery Extensions repo?

Comment (vineethvanga18) [reply ui/package.json:45]: Will do by today @leecalcote

**Comment by leecalcote** [comment]
@vineethvanga18 will you have this included in the meshery-extensions repo for Thursday's meeting?

**Pull request closed without merging**